wal_e/tar_partition.py

```diff
@@ -281,17 +281,13 @@ def tarfile_extract(fileobj, dest_path):
             assert not member.name.startswith('/')
             relpath = os.path.join(dest_path, member.name)
 
-            if member.isreg() and member.size >= pipebuf.PIPE_BUF_BYTES:
-                cat_extract(tar, member, relpath)
-            else:
-                tar.extract(member, path=dest_path)
-
             if member.issym():
-                # It does not appear possible to fsync a symlink, or
-                # so it seems, as there is no portable way to open()
-                # one to get a fd to run fsync on.
-                pass
+                os.symlink(member.linkname, os.path.join(dest_path, member.name))
             else:
+                if member.isreg() and member.size >= pipebuf.PIPE_BUF_BYTES:
```

Comment:
Why was this code restructured? It adds more code within branches when it seems no branch is necessary.